### ISSUE TITLE
Roles not returning

### ISSUE THREAD

**Issue by dark7622** [seanmcne/Microsoft.Xrm.Data.PowerShell#301]
So, odd issue, if I do a simple query like:

$roles = get-CRMRecords -EntityLogicalName role -conn $conn

I get nothing, yet there are plenty of roles, but I get no return.

The identity I'm using has permissions to Add CRM users via powershell, I'm trying to assign them a security role but I can't get any of the security roles to return. 

What would keep the roles from returning.

**Comment by seanmcne** [comment]
What is the fetch xml returned ($roles.FetchXml)?  You could run fiddler, as well, to see if there is a strange error or if there's some other issue that's not apparent - I'll try this out in an instance as well to see if I can repro but won't be able to test until I can get internet access on my PC (which may be later today).

**Comment by dark7622** [comment]
Fetch XML returns
<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false" no-lock="true"> 
<entity name="role"> 
<attribute name="roleid" /> 
</entity> 
</fetch> 

I will try Fiddler, but so far I can't get anything to return and no errors are popping

**Comment by dark7622** [comment]
"<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false" no-lock="true"> 
<entity name="role"> 
<attribute name="roleid" /> 
<filter type="and"> 
<condition attribute="name" operator="eq" value="{0}" /> 
<condition attribute="businessunitid" operator="eq" value="{1}" /> 
</filter> 
</entity> 
</fetch> "

**Comment by seanmcne** [comment]
try using three ticks at the top and three more ticks a the bottom for XML like this to use markdown for formatting

``` xml
<fetch version="1.0" output-format="xml-platform" mapping="logical" distinct="false" no-lock="true">
        <entity name="role">
            <attribute name='roleid' />
        </entity>
    </fetch>
```

Like this: 
![image](https://user-images.githubusercontent.com/2292260/39837383-f1fc1d02-539b-11e8-957a-0ebb681f357b.png)

**Comment by dark7622** [comment]
The query i got back looks exactly like what you gave as an example.

I am not using crm online, other queries work for getting users and creating users but for roles i get nothing.

**Comment by seanmcne** [comment]
Any chance you'd be willing to try xrm toolbox to run the fetch xml?

**Comment by dark7622** [comment]
I'm willing to try anything, downloaded XML toolbox and will try that out. Fiddler was a no go because of SSL issues. Though I didn't dive more into that just yet. Let me try this and report back.

**Comment by dark7622** [comment]
Update, Copy and paste the Fetch query to the FetchXML tester in the toolbox and it works. 

Fetches are working for other entities so I'm confused why it's not working for Role via the powershell framework.

**Comment by dark7622** [comment]
Issue resolved, working with the 8.2 version as well as a careful review have solved the problem.
